envoy/config/filter/http/ext_authz/v2/ext_authz.proto

```diff
@@ -23,4 +23,25 @@ message ExtAuthz {
   // is NOT denied then traffic will be permitted.
   // Defaults to false.
   bool failure_mode_allow = 2;
+
+  // When this value is set to true, the filter must call an HTTP authorization service only.
+  bool use_http_service = 3;
```

Comment:
I think an enum instead of a bool would allow future extensibility

Reply:
Better IMO would actually be to make this a `oneof` and just merge `HttpService` as one of the options inside the `oneof`. This is in general what we are trying to do elsewhere. This brings the enum and the "only use if enum is set" logic together. If this impacts the already implemented non-HTTP code I would just go ahead and change it now since IMO I still consider this filter under development.

Reply:
Unless I'm missing something, It seems that the impact on the already implemented non-HTTP code would be the same with either `enum` or `oneof`. 

The ExtAuthz message would look something like this: 
```
message ExtAuthz {

  oneof service_oneof {

    envoy.api.v2.core.GrpcService grpc_service = 1;

    HttpService http_service = 3;
  }

  bool failure_mode_allow = 2;
}
```

And the decision on which client to use, somewhere here: https://github.com/envoyproxy/envoy/blob/da0dc5f5dac12651f211e321f9cd7b2c70bedde9/source/server/config/http/ext_authz.cc#L17

Reply:
The `oneof` is a simpler way to represent mutually exclusive config when it's more than just a scalar value, i.e. the message. What you have there is basically what it would look like.